### PR TITLE
[osmium] Add package

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: O/osmium/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("osmium", :osmium)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="libosmium_jll", uuid="7f0a2ee1-9b43-5740-ae82-b54233cf7071"))
+    Dependency(PackageSpec(name="protozero_jll", uuid="e2028600-4f28-5e5c-ab86-957950af6e0a"))
+    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"))
```
Comment (giordano) [inline O/osmium/build_tarballs.jl:48]: Will I ever stop to need to link #3024? :slightly_smiling_face:

File: O/osmium/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("osmium", :osmium)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="libosmium_jll", uuid="7f0a2ee1-9b43-5740-ae82-b54233cf7071"))
```
Comment (giordano) [inline O/osmium/build_tarballs.jl:46]: Wasnt't `libosmium_jll` header-only?

File: O/osmium/bundled/patches/fix_windows_flag.patch [hunk 3]
```diff
@@ -0,0 +1,102 @@
+diff --git a/src/command_extract.cpp b/src/command_extract.cpp
+index e346b76..fa1d06c 100644
+--- a/src/command_extract.cpp
++++ b/src/command_extract.cpp
+@@ -68,7 +68,7 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ # include <io.h>
+ #endif
+ 
+-#ifndef _MSC_VER
++#if !defined(_MSC_VER) && !defined(_WIN32)
```
[the remark this answers is not in the document]
Comment (jeremiahpslewis) [reply O/osmium/bundled/patches/fix_windows_flag.patch]: Thanks

File: O/osmium/build_tarballs.jl (hunk 2 above)
Comment (jeremiahpslewis) [reply O/osmium/build_tarballs.jl]: Yup, thanks, fixed.

**Comment by jeremiahpslewis** [comment]
Upstream PR is here https://github.com/osmcode/osmium-tool/pull/254

File: O/osmium/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    -DCMAKE_BUILD_TYPE=Release
+
+make -j${nproc}
+make install
+"""
+
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline O/osmium/build_tarballs.jl:35]:
```
┌ Warning: /agent/_work/1/s/O/osmium/build/x86_64-linux-gnu/dodkMosJ/x86_64-linux-gnu-libgfortran3-cxx11/destdir/bin/osmium contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/dyIhw/src/auditor/compiler_abi.jl:247
```

**Pull request merged**